test/tests/utilities_internalTest.js

```diff
@@ -569,6 +569,45 @@ describe("Zotero.Utilities.Internal", function () {
 			assert.equal(out, '"" has a length of 0 and "hello" has a length of 5');
 		});
 
+		it("should support functions in comparison statements", function () {
+			const vars = {
+				sum: ({ a, b }) => parseInt(a) + parseInt(b),
+				fahrenheitToCelsius: ({ temp }) => Math.round((temp - 32) * 5 / 9),
+				false: 'false',
+				twoWords: 'two words',
+				onlyOne: 'actually == 1'
+			};
+			const template = `{{if sum a=1 b=2 == 3}}1 + 2 = {{sum a=1 b=2}}{{else}}no speak math{{endif}}`;
+			const out = Zotero.Utilities.Internal.generateHTMLFromTemplate(template, vars);
+			assert.equal(out, '1 + 2 = 3');
+
+			const template2 = '{{ if false>false }}no{{elseif false <= 0}}no{{elseif false != false}}no{{elseif false == false}}yes{{else}}no{{endif}}';
+			const out2 = Zotero.Utilities.Internal.generateHTMLFromTemplate(template2, vars);
+			assert.equal(out2, 'yes');
+
+			const template3 = '{{ if twoWords == two words }}yes{{else}}no{{endif}}';
+			const out3 = Zotero.Utilities.Internal.generateHTMLFromTemplate(template3, vars);
+			assert.equal(out3, 'yes');
+
+			const template4 = '{{ if onlyOne == actually == 1 }}yes{{else}}no{{endif}}';
```

Comment:
These last two don't seem good. I think we want to require single or double quotes for strings, as we do in the [documentation](https://www.zotero.org/support/note_templates). I don't really know what should happen for something like these, but it seems like a syntax error. Throw an error that gets logged?

And honestly I'm not sure we should even support anything other than string comparison with `==` and `!=`. This is a template engine, and the variables exist primarily to be interpolated into the final string. So I think even the `3` above should probably be `"3"`, and we should throw if a function returns something other than a string.

Maybe there'll be some reason for additional complexity in the future, and you could keep the code in a branch somewhere, but for now I feel like we should keep this as simple as possible.

Reply:
Some of these tests are to cover edge-cases and attempts at misuse the templating engine, I'd hope no one would use them like this.

> These last two don't seem good. I think we want to require single or double quotes for strings, as we do in the [documentation](https://www.zotero.org/support/note_templates).

I agree and would prefer that, however, current version does not require quotes for attributes, are we OK changing that?

> Maybe there'll be some reason for additional complexity in the future, and you could keep the code in a branch somewhere, but for now I feel like we should keep this as simple as possible.

I'll remove the numeric parsing and comparisons and throw if functions return anything else than a string.

Reply:
> I agree and would prefer that, however, current version does not require quotes for attributes, are we OK changing that?

Yes, let's require.

Reply:
I've tweaked the template engine so that single or double quotes are required for string literals. Without quotes, value is considered to be a "variable" and if not present in `vars` object, it will evaluate as empty string.